skyportal/utils/simsurvey.py

```diff
@@ -0,0 +1,95 @@
+import numpy as np
+from ..models import (
+    cosmo,
+)
+
+
+def get_simsurvey_parameters(model_name, optional_injection_parameters):
+    """Get the simsurvey injection model parameters
+    Parameters
+    ----------
+    model_name : str
+        Model to simulate efficiency for. Must be one of kilonova, afterglow, or linear. Defaults to kilonova.
+    optional_injection_parameters: dict
+        Optional parameters to specify the injection type, along with a list of possible values (to be used in a dropdown UI)
+    """
+
+    if model_name == "kilonova":
+        # default taken from https://github.com/mbulla/kilonova_models
+        default_parameters = {
+            "injection_filename": 'data/nsns_nph1.0e+06_mejdyn0.020_mejwind0.130_phi30.txt'
+        }
+    elif model_name == "afterglow":
+        default_parameters = {
+            "ntime": 30,
+            "t_i": 0.25,
+            "t_f": 15.25,
+            "nlambda": 100,
+            "lambda_min": 1000,
+            "lambda_max": 25000,
+            "jetType": 0,
+            "specType": 0,
+            "thetaObs": 0.0,
+            "log10_E0": 53.0,
+            "thetaCore": np.pi / 4.0,
+            "thetaWing": np.pi / 4.0,
+            "b": 0.0,
+            "log10_n0": 1.0,
+            "p": 2.5,
+            "log10_epsilon_e": -1,
+            "log10_epsilon_B": -2,
+            "d_L": 3.09e19,
+            "xi_N": 1.0,
+            "z": 0.0,
+        }
+    elif model_name == "linear":
+        default_parameters = {
+            "ntime": 30,
+            "t_i": 0.25,
+            "t_f": 15.25,
+            "nlambda": 100,
+            "lambda_min": 1000,
+            "lambda_max": 25000,
+            "mag": -16.0,
+            "dmag": 1.0,
+        }
+    # for when we use python 3.9 by default
+    # optional_injection_parameters = default_parameters | optional_injection_parameters
+    optional_injection_parameters = {
+        **default_parameters,
+        **optional_injection_parameters,
+    }
```

Comment:
I would replace `default_parameters` with just `parameters` then `parameters.update(optional_injection_parameters` and then use `parameters` instead of `optional_injection_parameters` below.

Reply:
yes this is more clear.